src/AuthorizedSocketConnection.js

```diff
@@ -176,11 +179,18 @@ export default class AuthorizedSocketConnection<TContext, TCredentials> {
         document,
         null,
         {
-          subscribe: async (...args) =>
-            AsyncUtils.filter(
-              await subscriptionContext.subscribe(...args),
-              this.isAuthorized,
-            ),
+          subscribe: async (
+            channel,
+            { authorizationPredicate, ...options },
+          ) => {
+            return AsyncUtils.authorizationFilter(
+              await subscriptionContext.subscribe(channel, options),
+              this.getAuthorizationChecker(
+                authorizationPredicate || this.authorizationPredicate,
+              ),
+              this.config.credentialsManager.getCredentials(),
```

Comment:
looks like you've got the start two slightly different implementations mixed up here. 

- `getAuthorizationChecker` uses a closure to pass in `credentials`,
- `authorizationFilter()`, filters and also passes in credentials

If i'm reading this right...you can do one or the other, but don't need both. Personally i'd drop `authorizationFilter()` for the plain `filter` and use the `getAuthorizationChecker` to pass in the credentials.

Reply:
GOOD POINT